### PR TITLE
Implemented movie discover & Add Original to AssetQuality

### PR DESCRIPTION
**Pull request by Akash98Sky**
Only tv shows left now :)

### CODE REVIEW

File: lib/src/models/date.dart [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  }) : super(year, month, day);
+
+  static Date tryParse(String formattedString) {
+    if (formattedString == null || formattedString.isEmpty) return null;
+
+    DateTime dateTime = DateTime.tryParse(formattedString);
+    return Date(day: dateTime.day, month: dateTime.month, year: dateTime.year);
+  }
+
+  static String _fourDigits(int n) {
```
Comment (Kerego) [inline lib/src/models/date.dart]: This could be rewritten using padleft(...)

File: lib/src/models/date.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    int absN = n.abs();
+    String sign = n < 0 ? "-" : "";
+    if (absN >= 1000) return "$n";
+    if (absN >= 100) return "${sign}0$absN";
+    if (absN >= 10) return "${sign}00$absN";
+    return "${sign}000$absN";
+  }
+
+  static String _twoDigits(int n) {
```
Comment (Kerego) [inline lib/src/models/date.dart]: This could be rewritten using padleft(...)

File: lib/src/services/movie-service.dart [hunk 3]
```diff
@@ -69,9 +91,12 @@ abstract class MovieService {
   Future<Response> getWithResilience(Uri uri);
 
   Future<PagedResult<MovieBase>> _fetchPagedResult(
-      MovieSearchSettings settings, String url) async {
+    MovieSearchSettings settings,
+    String url,
+  ) async {
     var queryParams = settings.toJson()..["api_key"] = _apiKey;
     Uri uri = _buildUri(url, queryParams);
+    print(uri.toString());
```
Comment (Kerego) [inline lib/src/services/movie-service.dart]: this is probably redundant

File: lib/src/services/movie-service.dart [hunk 4]
```diff
@@ -35,14 +39,32 @@ abstract class MovieService {
...
       _fetchPagedResult(settings, "3/movie/latest");
 
-  Future<Movie> getMovie(int id,
-      {String language = "en-US",
-      List<String> imageLanguages = const ["en", null],
-      MovieAppendSettings appendSettings,
-      QualitySettings qualitySettings}) async {
-    appendSettings = appendSettings ?? MovieAppendSettings();
-    qualitySettings = qualitySettings ?? QualitySettings();
+  Future<List<Genre>> getAllMovieGenres({String language = "en-US"}) async {
```
Comment (Kerego) [inline lib/src/services/movie-service.dart]: I would move this to configuration service, as requests that could be cached are better suited to sit there.

File: lib/src/settings/configuration.dart [hunk 5]
```diff
@@ -32,13 +32,26 @@ class Configuration {
...
+  }) {
+    assert(baseUrl != null);
+    assert(secureBaseUrl != null);
+    assert(backdropSizes != null && backdropSizes.isNotEmpty);
+    assert(logoSizes != null && logoSizes.isNotEmpty);
+    assert(posterSizes != null && posterSizes.isNotEmpty);
+    assert(profileSizes != null && profileSizes.isNotEmpty);
+    assert(stillSizes != null && stillSizes.isNotEmpty);
+
+    backdropSizes.sort((v1, v2) =>
```
[the remark this answers is not in the document]
Comment (Akash98Sky) [reply lib/src/settings/configuration.dart]: Yes, thought of it. But found no better solution. As there's manual configuration also so I added this.

File: example/main.dart [hunk 6]
```diff
@@ -18,11 +18,11 @@ Future main(List<String> arguments) async {
...
     baseUrl: "http://image.tmdb.org/t/p/",
     secureBaseUrl: "https://image.tmdb.org/t/p/",
-    backdropSizes: ["original"],
-    logoSizes: ["original"],
-    posterSizes: ["original"],
-    profileSizes: ["original"],
-    stillSizes: ["original"],
+    backdropSizes: ["w300", "w780", "w1280", "original"],
+    logoSizes: ["w300", "w780", "w1280", "original"],
+    posterSizes: ["w300", "w780", "w1280", "original"],
```
[the remark this answers is not in the document]
Comment (Akash98Sky) [reply example/main.dart]: ok.

File: lib/src/services/movie-service.dart (hunk 3 above)
Comment (Akash98Sky) [reply lib/src/services/movie-service.dart]: Ahh sorry. I was using it while debugging and forgot to remove later. 

I'll delete it.

File: lib/src/services/movie-service.dart (hunk 4 above)
Comment (Akash98Sky) [reply lib/src/services/movie-service.dart]: Yeah that should be fine.

File: lib/src/settings/configuration.dart (hunk 5 above)
[the remark this answers is not in the document]
Comment (Akash98Sky) [reply lib/src/settings/configuration.dart]: But I think in auto configure there's no h and w representation mixed in a single list.
If so then this way of sorting is fine.
Or if I add another optional parameter {bool sortingNeeded = false}.

[the remark this answers is not in the document]
Comment (Kerego) [reply lib/src/settings/configuration.dart]: The profile sizes has both mixed up, which results in the following:
image

[the remark this answers is not in the document]
Comment (Akash98Sky) [reply lib/src/settings/configuration.dart]: Okay. Thanks for clarifying.

**Pull request merged**